### PR TITLE
Use react error boundary to gracefully catch rendering bugs

### PR DESCRIPTION
**Pull request by philippotto**
At the moment, an error within a react component will simply crash wk. When developing, the page instantly refreshes without a chance to see the actual error (when dev tools are closed). If the error re-occurs on render, this causes an endless loop.

This PR fixes that issue by using a react error boundary, which catches rendering bugs and shows a toast message.

### Steps to test:
- ensure that normal rendering is unaffected
- introduce an error somewhere in a component (throw new Error in a render() method) and see that a toast is displayed

------
- [X] Ready for review

### CODE REVIEW

File: app/assets/javascripts/oxalis/view/tracing_layout_view.js [hunk 1]
```diff
@@ -44,6 +45,12 @@ class TracingLayoutView extends React.PureComponent<Props, State> {
     });
   };
 
+  componentDidCatch() {
+    Toast.error(
+      "Unfortunateley, we encountered an error. We cannot guarantee that your work is persisted. Please reload the page and try again.",
```
Comment (philippotto) [inline app/assets/javascripts/oxalis/view/tracing_layout_view.js]: @hotzenklotz Apart from the "Unfortunateley" typo, do you have suggestions to improve this message, mr. copy master?

Comment (normanrz) [reply app/assets/javascripts/oxalis/view/tracing_layout_view.js]: Also, please move to messages.js

File: app/assets/javascripts/oxalis/view/tracing_layout_view.js [hunk 2]
```diff
@@ -44,6 +45,12 @@ class TracingLayoutView extends React.PureComponent<Props, State> {
     });
   };
 
+  componentDidCatch() {
+    Toast.error(
```
Comment (daniel-wer) [inline app/assets/javascripts/oxalis/view/tracing_layout_view.js]: We could `console.error` before the toast, so the error would be visible in the console as well (no need to break).

Comment (philippotto) [reply app/assets/javascripts/oxalis/view/tracing_layout_view.js]: The error is logged either way, so that shouldn't be necessary :)

**Pull request merged**